src/routes/TradingLeague/TradingLeague.tsx

```diff
@@ -32,26 +34,24 @@ const TradingLeague: React.FunctionComponent = () => {
 
   const fetchRankings = async () => {
     setLoading(true);
-    const result = await getActivity({
-      from: season.startDate.toSeconds(),
-      ignoredWalletIds:
-        process.env.REACT_APP_IGNORED_LEAGUE_WALLETS &&
-        process.env.REACT_APP_IGNORED_LEAGUE_WALLETS !== 'UNPROVIDED'
-          ? process.env.REACT_APP_IGNORED_LEAGUE_WALLETS.split(',')
-              .map((s) => s.trim().toLowerCase())
-              .reduce((pV, cI) => ({ ...pV, [cI]: true }), {})
-          : {},
-    });
+    let result: RankType[] = [];
+    const params = getSDKInitParams(signer);
+    if (params) {
+      const SBT = new CommunitySBT(params);
+      result = await SBT.getRanking({
+        seasonStart: season.startDate.toSeconds(),
+        seasonEnd: season.endDate.toSeconds(),
+      });
+    }
     setUserRanking(result, wallet.account);
     setRankings(result);
     setLoading(false);
   };
 
   const setUserRanking = (rankingResults: RankType[], walletAddress?: string | null) => {
-    const userEntry: RankType | undefined = rankingResults.find((r) => r.address === walletAddress);
-
+    const userEntry: RankType | undefined = rankingResults.find((r) => r.address.toLowerCase() === walletAddress?.toLowerCase());
     setUserPoints(userEntry?.points || -1);
-    setUserRank(userEntry?.rank || -1);
+    setUserRank( userEntry ? (isUndefined(userEntry.rank) ? -1 : userEntry.rank) : -1);
```

Comment:
how is this different to ```    setUserRank(userEntry?.rank || -1);```

Reply:
userEntry.rank can be undefined even if is defined

Reply:
but this code; setUserRank(userEntry?.rank || -1); already takes care of that. If undefined I default it to -1. not sure why we need to complicate it with isUndefined logic

Reply:
oh, my bad. so the issue here is that if rank === 0 then it returns -1 but it should actually be #1. not sure what the logic behind this is but the code above seems to fix the problem of the user's profile showing rank -- when they are actually first.

Reply:
then setUserRank(userEntry?.rank ?? -1) is much nicer and cleaner way to solve it. Nice catch for the bug

Reply:
7499fabd replaced with proposed `userEntry?.rank ?? -1`